Fixes #20150 by updating global-json.md to have cleaner info on match… (#21290)
dotnet/docs: docs/core/tools/global-json.md

- Available since: .NET Core 3.0 SDK.

The roll-forward policy to use when selecting an SDK version, either as a fallback when a specific SDK version is missing or as a directive to use a higher version. A [version](#version) must be specified with a `rollForward` value, unless you're setting it to `latestMajor`.

> The default value roll forward behavior is determined by the [matching rules](#matching-rules).

Reviewer: Looks like I accidentally left an extra word in my earlier suggestion.

```suggestion
The default roll forward behavior is determined by the [matching rules](#matching-rules).
```

> The default value of `rollForward` is determined by the [matching rules](https://docs.microsoft.com/dotnet/core/tools/global-json?tabs=netcore3x#matching-rules).

Reviewer 2: ```suggestion
The default value of `rollForward` is determined by the [matching rules](#matching-rules).
```

Author: Thanks!

- Available since: .NET Core 3.0 SDK.

The roll-forward policy to use when selecting an SDK version, either as a fallback when a specific SDK version is missing or as a directive to use a higher version. A [version](#version) must be specified with a `rollForward` value, unless you're setting it to `latestMajor`.

> The default value of `rollForward` is determined by the [matching rules](#matching-rules).

Reviewer: ```suggestion
The default value roll forward behavior is determined by the [matching rules](#matching-rules).
```

Author: Thanks!

## Matching rules

> [!NOTE]

> > The matching rules are governed by the `dotnet.exe` entry point, which is common across all installed .NET Core installed runtimes. The matching rules for the latest installed version of the .NET Core Runtime are used when you have multiple runtimes installed side-by-side or if or you're using a global.json.

Reviewer: ```suggestion
> The matching rules are governed by the `dotnet.exe` entry point, which is common across all installed .NET Core installed runtimes. The matching rules for the latest installed version of the .NET Core Runtime are used when you have multiple runtimes installed side-by-side or if or you're using a *global.json* file.
```
